### DISCUSSION TITLE
Sandbox to test out new features and nfts

### DISCUSSION THREAD

**Discussion by drumdrumdrum3337** [sunflower-land/sunflower-land#1502] [Ideas]
**Description**

Very soon the game moves out from Beta into the wild, which means that retracting wrong steps, abandoning paths taken, and making changes overall would become much more complicated. I imagine, in simpler games it is easier to test within the small dev team, but as the Sunflower Land enters tricky waters of real economy, having a space to safely test various approaches might be crucial for longterm success.

**What is Sandbox**

The Sandbox is the exact copy of the game that could be reached by a certain set of players in order to test some change, feature, or an nft before it gets released in order to assess its impact on the game.

Basic functionality of the Sandbox:

- Sandbox is available to a defined set of players whom devs trust to provide good feedback;
- For each player, Sandbox is launched by making an exact copy of the player's farm on the day when the testing is supposed to start;
- From then on, player plays both the regular game and the sandbox as though these are 2 separate games, utilizing the new features in the sandbox;
- Number of Sandboxes don't have to be limited to 1: we could be interested to assess the impact of a variation of FB nft (and do it for 8 weeks), and at the same time we want to see what economy holds better in the long run - deflationary or inflationary, and run another test for 6 months.

Important notes:
- I have no idea how this might work technically, but since there's no need for "real" syncs it's probably not as complicated to develop?
- Results from Sandboxes could be indeed just one input into the overall decision process;
- Each test would require proper planing with limitations discussed in advance (e.g. if testing is done with 2k users the market might behave differently than with 1mil users indeed).

This is indeed a very high level idea, and it might not hold well against detailed scrutiny, but still might be worth discussing.

**Comment by adamhannigan** [comment]
@drumdrumdrum3337 Nice write up 👏 

Could you provide some more details how this would differ from our testnet?

It is great to do testing with A/B style using feature flags. Our approach to new features will likely go towards being in the same app but we feature flag it (toggle on and off) for a sample size of users (1-2K users). These would essentially be our 'new feature testers', but the gameplay itself would still appear under the same domain 😄

**Reply by drumdrumdrum3337** [reply]
@adamhannigan well, I have only assumptions about the testnet, such as that it is an environment where you can test a variety of game features that are being developed, spot issues, etc. To play with the game, so to speak. However, it is more of right here right now type of testing.

When suggesting the sandbox, I wasn't thinking as much about A/B testing or the like, but rather about modelling more complex situations to see how they would affect the game in the long run. I was thinking more about the time when the game really takes off - economy will inevitably start to lean towards one direction or the other, and you'd have to take some actions to put it back into balance. 

A simplistic example: there's too much wood in the game to the point that it is being sold cheaper than the cost of production. Wood burning event might help, but how to decide what would be the main prize for burning it? Would a rare but decoration nft work for folks, or they would say 'nah' and ignore the event? Is it going to be some boost nft, and if yes, then what the actual boost would be? What effect on the economy it might have? How might community react? And so on.

From my laic point of view, being able to test out such various actions in a safe environment and on all types of players (recent ones, elders, whales, those with no investments, etc) might be useful. I believe economists run various models to understand the impact of suggested changes.

That said, I have no idea about the decision process on your end, so just throwing this idea on the table with complete understanding that it might be irrelevant, or just sound good in theory but completely impractical in reality.
